examples/plot_firing_pattern.py

```diff
@@ -100,7 +100,7 @@
 ###############################################################################
 # We can plot the firing pattern of individual cells by indexing with the gid
 gid = 170
-plt.figure(figsize=(4, 4))
+plt.figure(figsize=(4, 4), constrained_layout=True)
```

Comment:
Maybe use `plt.tightlayout()` instead? It seems that this setting may break our figures in a matplotlib update:

![image](https://user-images.githubusercontent.com/55253912/108375146-b4c65d80-71cf-11eb-8700-bbf2c63b33be.png)

Reply:
unfortunately this doesn't work for more complicated plots (like the one in `plot_simulate_gamma`). Without some extra effort, many of our plots get the ylabel cut out (see e.g. the firing rate demo in our master docs). The challenge is to find something that respects `sharex=True`, *and* allows variable-width labels (1-2 lines), **and** understands colorbars... Any suggestions welcome! (`tight_layout` doesn't cut it)

I wonder if we need to worry about that warning though? I think `hnn-core` will be developing a lot faster than `matplotlib`, so I'm sure we'll come you with new plots faster than they can remove functionality from us ;)

Reply:
can't we just get rid of this argument? Does it look too bad? The aim of the examples is not to make beautiful figures. People can always add `plt.tight_layout` to their scripts.

Reply:
But the warnings don't look too bad. I am fine with letting it stay :)